raiden/channel.py

```diff
@@ -534,6 +534,10 @@ def __init__(self, our_state, partner_state, external_state,
         self.settle_timeout = settle_timeout
         self.external_state = external_state
 
+        # use nonce ranges
+        self.our_state.nonce = our_state.nonce * (external_state.opened_block * (2 ** 32))
+        self.partner_state.nonce = partner_state.nonce * (external_state.opened_block * (2 ** 32))
```

Comment:
The constructor is *the* place to initialize an object, without passing `nonce` in the constructor we end up with a object improperly initialized (eg. it will produce the wrong behavior)